eclipse-collections-code-generator/src/main/resources/impl/collection/mutable/abstractMultiReaderPrimitiveCollection.stg

```diff
@@ -0,0 +1,554 @@
+import "copyright.stg"
+import "primitiveEquals.stg"
+import "primitiveHashCode.stg"
+import "primitiveLiteral.stg"
+
+targetPath() ::= "org/eclipse/collections/impl/collection/mutable/primitive"
+
+fileName(primitive) ::= "AbstractMultiReader<primitive.name>Collection"
+
+class(primitive) ::= <<
+<body(primitive.type, primitive.name)>
+>>
+
+readLock() ::= <<
+this.acquireReadLock();
+try
+{
+>>
+
+writeLock() ::= <<
+this.acquireWriteLock();
+try
+{
+>>
+
+readUnlock() ::= <<
+}
+finally
+{
+    this.unlockReadLock();
+}
+>>
+
+writeUnlock() ::= <<
+}
+finally
+{
+    this.unlockWriteLock();
+}
+>>
+
+body(type, name) ::= <<
+<copyright()>
+
+package org.eclipse.collections.impl.collection.mutable.primitive;
+
+import java.io.Serializable;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.eclipse.collections.api.<name>Iterable;
+import org.eclipse.collections.api.Lazy<name>Iterable;
+import org.eclipse.collections.api.bag.primitive.Mutable<name>Bag;
+import org.eclipse.collections.api.block.function.primitive.<name>ToObjectFunction;
+import org.eclipse.collections.api.block.function.primitive.Object<name>ToObjectFunction;
+import org.eclipse.collections.api.block.predicate.primitive.<name>Predicate;
+import org.eclipse.collections.api.block.procedure.primitive.<name>Procedure;
+import org.eclipse.collections.api.collection.MutableCollection;
+import org.eclipse.collections.api.collection.primitive.Immutable<name>Collection;
+import org.eclipse.collections.api.collection.primitive.Mutable<name>Collection;
+import org.eclipse.collections.api.iterator.Mutable<name>Iterator;
+import org.eclipse.collections.api.iterator.<name>Iterator;
+import org.eclipse.collections.api.list.primitive.Mutable<name>List;
+import org.eclipse.collections.api.set.primitive.Mutable<name>Set;
+import org.eclipse.collections.impl.lazy.primitive.Lazy<name>IterableAdapter;
+import net.jcip.annotations.ThreadSafe;
+
+/**
+ * This file was automatically generated from template file abstractMultiReaderPrimitiveCollection.stg.
+ *
+ * @since 8.1.
+ */
+@ThreadSafe
+public abstract class AbstractMultiReader<name>Collection
+        implements Mutable<name>Collection, Serializable
```

Comment:
Lately I've been trying to avoid making abstract classes Serializable. The serialized form of the subclass will include information about the superclass, increasing the size of the serialized form in bytes and making additional refactoring more likely to break the serialized form.